pkg/registry/memory/ns_memory_server.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package memory
+
+import (
+	"context"
+
+	"github.com/networkservicemesh/api/pkg/api/registry"
+	"github.com/pkg/errors"
+)
+
+type memoryServiceDiscoveryServer struct {
+	storage *Storage
+}
+
+func (d *memoryServiceDiscoveryServer) FindNetworkService(ctx context.Context, req *registry.FindNetworkServiceRequest) (*registry.FindNetworkServiceResponse, error) {
+	service, ok := d.storage.NetworkServices.Load(req.NetworkServiceName)
```

Comment:
should we also use spanhelper here, we start to forget to use spans :-)

Reply:
@edwarnicke

Reply:
@haiodo Can we just use chain element `pkg/trace` to adding spans?

Reply:
@haiodo I wouldn't use spanhelper *in* the chain element... I do think having a chain element that uses spanhelper (as we do in our use of trace for NetworkService) is goodness though.  Basically... the only code where we should *see* spanhelper is the trace chain elements :)

Reply:
Ok, sounds good to me.